### DISCUSSION TITLE
How to update?

### DISCUSSION THREAD

**Discussion by DandeMC** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#267] [Q&A]
Hello,

I'm on version 2.1.2 and I want to update. I now the upload.html but which file do I have to select for the update?

**Comment by DandeMC** [comment]
Ok. I must upload it via platform.io in visual studio code. but when I have done this, it resets my mqtt settings, but the interface is the same as the old one. Can anyone help me?

**Reply by visualapproach** [reply]
Basically the same procedure as the first install but you can upload over the air. The filesystem is probably easier to do file by file to preserve settings. Use the web ui to upload the files in the data/ folder.

**Reply by DandeMC** [reply]
Which web ui do you mean?

**Reply by DandeMC** [reply]
2 problems since I uploaded the now file OTA from platform.io:
- I can't save MQTT settings and it's blank
- It shows the hydrojets slider although I have #define PRE2021 active

**Reply by visualapproach** [reply]
The web ui of the ESP:
![image](https://user-images.githubusercontent.com/22987250/167673613-da3dd4f3-08b5-46be-ab0e-a6951e9c826c.png)

MQTT dialog is empty a few seconds while loading the data. Perhaps you did something to the file system?
It doesn't work sometime to upload filesystem over the air. Either do it via USB or just upload the files one by one with the upload menu above.

**Reply by DandeMC** [reply]
I've installed it complete new over usb. It was the clean way. And now it works fine :-)

**Reply by visualapproach** [reply]
Nice 👍🏼
